specs/_features/eip7594/polynomial-commitments-sampling.md

```diff
@@ -527,9 +529,13 @@ def verify_cell_kzg_proof_batch_impl(row_commitments: Sequence[KZGCommitment],
 
     # Step 4: Compute RL = RLC - RLI + RLP
     # Step 4.1: Compute RLC = sum_i (sum_{k_i} r^{k_i}) commitments[i]
+    # Step 4.1a: Compute commitment_weights[i] = (sum_{k_i} r^{k_i}) for all rows i
+    # Note: we do that by iterating over all k_i and updating commitment_weights[i] accordingly
     commitment_weights = [0] * num_rows
-    for k in range(num_cells):
-        commitment_weights[row_indices[k]] = (commitment_weights[row_indices[k]] + int(r_powers[k])) % BLS_MODULUS
+    for k_i in range(num_cells):
```

Comment:
Thought a bit more about it. The notation on the blog post is confusing.

Regardless of the blog post confusion, `k_i` above is *documented* as `k_i < len(row_indices) ranges over all cells that are in row i`, but I don't think this is the case for the code *variable* `k_i` which ranges over all the cells.

I agree that `i = row_indices[k]` based on the definition of `i` above.
I also agree that `commitment_weights[i] += r^k` based on the definitions of `i` and `k`.

IMO we should kill the concept of `k_i` from the code and comments because it's confusing (see below), and do everything in terms of `i` and `k`.

In particular, the way `k_i` right now is defined is that it "ranges over all cells", and usually when I hear "ranges" I think that it's a list that goes from 0 to the number of cells in row `i`. But that's not the case, instead `k_i` is a list which contains the indices of cells of row `i`. However, we never really build such a list in the code, or even use `k_i` in such a way, so perhaps we should just remove it.

Alternatively, let's fix the documentation of `k_i` and provide an explanation of how we go from `k` and `i` to `k_i`.